### PR TITLE
Implement new parsing for of XS in exciting

### PR DESCRIPTION
**Pull request by ladinesa**

### CODE REVIEW

**Comment by JosePizarro3** [comment]
@ladinesa please, mainly check the new test + `parse_xs_worklfow`. Once we are happy, we can merge it and go to nomad for finishing the references and metainfo.

Did you already checked in there? Maybe you can and once it is done it is a matter of rebasing and updating the ref to electronicparsers.

File: electronicparsers/exciting/parser.py [hunk 1]
```diff
@@ -1594,164 +1577,320 @@ def parse_epsilon(data, sec_scc):
...
+        self.archive.workflow2 = workflow
+
+    def parse_xs_worklfow(self, xs_archive, xs_workflow_archive):
+        sec_run = xs_workflow_archive.m_create(Run)
+        sec_run.program = self.archive.run[-1].program
+        sec_run.system = self.archive.run[-1].system
+        workflow = ParticleHoleExcitations(
+            results=ParticleHoleExcitationsResults(), method=ParticleHoleExcitationsMethod())
+
+        def extract_section(source, path):
```
Comment (ladinesa) [inline electronicparsers/exciting/parser.py]: make this a class method as used also in GW

File: electronicparsers/exciting/parser.py [hunk 2]
```diff
@@ -1594,164 +1577,320 @@ def parse_epsilon(data, sec_scc):
...
+            if len(path_segments) == 1:
+                return value
+            else:
+                return extract_section(value, path_segments[1])
+
+        def extract_polarization_outputs():
+            output = []
+            index = 0
+            for path, archive in self._child_archives.items():
+                if 'EPSILON' in path:
```
Comment (ladinesa) [inline electronicparsers/exciting/parser.py]: should be `if os.path.basename(path).split('_')[0] in self._spectra_types`

File: electronicparsers/exciting/parser.py [hunk 3]
```diff
@@ -1594,164 +1577,320 @@ def parse_epsilon(data, sec_scc):
...
+        workflow.outputs = extract_polarization_outputs()
+        input_calculation = extract_section(self.archive, 'run/calculation')
+        if self.archive.workflow2:  # DFT task
+            task = TaskReference(task=self.archive.workflow2)
+            if input_structure:
+                task.inputs = [Link(name='Input structure', section=input_structure)]
+                if input_calculation:
+                    task.outputs = [
+                        Link(name='Input structure', section=input_structure),
+                        Link(name='Output DFT calculation', section=input_calculation)]
```
Comment (ladinesa) [inline electronicparsers/exciting/parser.py]: can you probably make this a variable e.g. name_output_calc = 'Output DFT calculation' so it is easier to change it

**Comment by ladinesa** [comment]
I think it is all good. please merge

File: electronicparsers/exciting/parser.py (hunk 1 above)
Comment (JosePizarro3) [reply electronicparsers/exciting/parser.py]: done

File: electronicparsers/exciting/parser.py (hunk 2 above)
Comment (JosePizarro3) [reply electronicparsers/exciting/parser.py]: done

File: electronicparsers/exciting/parser.py (hunk 3 above)
Comment (JosePizarro3) [reply electronicparsers/exciting/parser.py]: done

**Pull request merged**